framework/console/widgets/Table.php

```diff
@@ -0,0 +1,323 @@
+    ];
+    /**
+     * @var array table column widths
+     */
+    private $_columnWidths = [];
+    /**
+     * @var integer screen size
+     */
+    private $_screenSize;
+    /**
+     * @var string list prefix
+     */
+    private $_listPrefix = '• ';
+
+    /**
+     * Set table headers
+     *
+     * @param array $headers table headers
+     * @return $this
+     */
+    public function setHeaders(array $headers)
+    {
+        $this->_headers = $headers;
+        return $this;
+    }
+
+    /**
+     * Set table rows
+     *
+     * @param array $rows table rows
+     * @return $this
+     */
+    public function setRows(array $rows)
+    {
+        $this->_rows = $rows;
+        return $this;
+    }
+
+    /**
+     * Set table chars
+     *
+     * @param array $chars table chars
+     * @return $this
+     */
+    public function setChars(array $chars)
+    {
+        $this->_chars = $chars;
+        return $this;
+    }
+
+    /**
+    * Set screen width
+    *
+    * @param int $width screen width
+    * @return $this
+    */
+    public function setScreenSize($width)
+    {
+        $this->_screenSize = $width;
+        return $this;
+    }
+
+    /**
+     * Set list prefix
+     *
+     * @param string $listPrefix list prefix
+     * @return $this
+     */
+    public function setListPrefix($listPrefix)
+    {
+        $this->_listPrefix = $listPrefix;
+        return $this;
+    }
+
+    /**
+     * @return string the generated table
+     */
+    public function render()
+    {
+        $this->calculateSizeRows();
+        $buffer = $this->renderSeparator(
+            $this->_chars['top-left'],
+            $this->_chars['top-mid'],
+            $this->_chars['top'],
+            $this->_chars['top-right']
+        );
+        $buffer .= $this->renderRows($this->_headers,
+            $this->_chars['left'],
+            $this->_chars['middle'],
+            $this->_chars['right']
+        );
+
+        foreach ($this->_rows as $row) {
+            $buffer .= $this->renderSeparator(
+                $this->_chars['left-mid'],
+                $this->_chars['mid-mid'],
+                $this->_chars['mid'],
+                $this->_chars['right-mid']
+            );
+            $buffer .= $this->renderRows($row,
+                $this->_chars['left'],
+                $this->_chars['middle'],
+                $this->_chars['right']);
+        }
+
+        $buffer .= $this->renderSeparator(
+            $this->_chars['bottom-left'],
+            $this->_chars['bottom-mid'],
+            $this->_chars['bottom'],
+            $this->_chars['bottom-right']
+        );
+
+        return $buffer;
+    }
+
+    /**
+     * @param array $row
+     * @param string $spanLeft
+     * @param string $spanMiddle
+     * @param string $spanRight
+     * @return string the generated row
+     * @see \yii\console\widgets\Table::render()
+     */
+    protected function renderRows(array $row, $spanLeft, $spanMiddle, $spanRight)
```

Comment:
It will be great to split this method to smaller ones to decrease its complexity. The method has some parts that do some work and may be names somehow